CONTRIBUTING.md

```diff
@@ -1,18 +1,102 @@
- Just found out about the [Zen of Python](https://www.python.org/dev/peps/pep-0020/) which might be a good guide in general
-```
-Beautiful is better than ugly.
-Explicit is better than implicit.
-Simple is better than complex.
-Complex is better than complicated.
-Flat is better than nested.
-Sparse is better than dense.
-Readability counts.
-Special cases aren't special enough to break the rules.
-Although practicality beats purity.
-Errors should never pass silently.
-In the face of ambiguity, refuse the temptation to guess.
-There should be one-- and preferably only one --obvious way to do it.
-Now is better than never.
-If the implementation is hard to explain, it's a bad idea.
-If the implementation is easy to explain, it may be a good idea.
-```
+# Contributors Guide
+
+Thank you for considering contributing to Oceananigans! This short guide will
+give you ideas on how you can contribute and help you make a contribution.
+
+Please feel free to ask us questions and chat with us at any time if you're
+unsure about anything.
+
+## What can I do?
+
+* Tackle an existing issue. We keep a list of [good first issues](https://github.com/climate-machine/Oceananigans.jl/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
+  that are self-contained and suitable for a newcomer to try and work on.
+
+* Try to run Oceananigans and play around with it to simulate your favorite
+  fluids and ocean physics. If you run into any problems or find it difficult
+  to use or understand, please open an issue!
+
+* Write up an example or tutorial on how to do something useful with
+  Oceananigans, like how to set up a new physical configuration.
+
+* Improve documentation or comments if you found something hard to use.
+
+* Implement a new feature if you need it to use Oceananigans.
+
+If you're interested in working on something, let us know by commenting on
+existing issues or by opening a new issue if. This is to make sure no one else
+is working on the same issue and so we can help and guide you in case there
+is anything you need to know beforehand.
+
+## Ground Rules
+
+* Each pull request should consist of a logical collection of changes. You can
+  include multiple bug fixes in a single pull request, but they should be related.
+  For unrelated changes, please submit multiple pull requests.
+* Do not commit changes to files that are irrelevant to your feature or bugfix
+  (eg: .gitignore).
+* Be willing to accept criticism and work on improving your code; we don't want
+  to break other users' code, so care must be taken not to introduce bugs. We
+  discuss pull requests and keep working on them until we believe we've done a
+  good job.
+* Be aware that the pull request review process is not immediate, and is
```

Comment:
Should we add a statement that we encourage either commenting on an existing issue or opening a new one to discuss the problem being tackled prior to starting work on the PR?